smart-contracts/test/Unlock/UnlockProxy.js

```diff
@@ -0,0 +1,72 @@
+const zos = require('zos-lib')
+const Unlock = artifacts.require('Unlock')
+const UnlockTestV2 = artifacts.require('UnlockTestV2')
+const AdminUpgradeabilityProxy = zos.Contracts.getFromNodeModules('zos-lib', 'AdminUpgradeabilityProxy')
+const shared = require('./behaviors/shared')
+const Units = require('ethereumjs-units')
+
+contract('Unlock', function (accounts) {
+  const proxyOwner = accounts[1]
+  const unlockOwner = accounts[2]
+
+  describe('Proxy Unlock contract', function () {
+    before(async function () {
+      const implV1 = await Unlock.new()
+      this.proxy = await AdminUpgradeabilityProxy.new(implV1.address, { from: proxyOwner })
+      this.unlock = await Unlock.at(this.proxy.address)
+      await this.unlock.initialize(unlockOwner)
+    })
+
+    describe('should function as a proxy', function () {
+      shared.shouldBehaveLikeV1(accounts, unlockOwner)
+
+      it('should fail if called from the proxy owner\'s account', async function () {
+        try {
+          await this.unlock.grossNetworkProduct({from: proxyOwner})
```

Comment:
Do you mind explaining why this is the case?

Reply:
I'm not able to explain the rational.  Not sure I understand.  However you can see this is driven by line 127 of their AdminUpgradeabilityProxy:

```
  function _willFallback() internal {
    require(msg.sender != _admin(), "Cannot call fallback function from the proxy admin");
    super._willFallback();
  }
```

Reply:
Hum. Ok. I think I need to investigate this.